nasa-gibs/worldview
# WV-2177: Replacing the Redux-Responsive library for better mobile detection

## Description

Before this fix, we were using the Redux-Responsive library to update the app whenever the screen size changed. This library has some drawbacks such as a lack of ability to detect mobile devices. This meant that sometimes when changing the orientation on some mobile devices, the app would render in a desktop view instead of a mobile view. This would happen because the Redux-Responsive library would detect that the screen width was now wider than our breakpoints for 'medium' and render as a desktop view. 

To correct this I have replaced most of the Redux-Responsive capabilities with our own actions and reducers. I also implemented a smaller library called 'react-device-detect' that parses the user agent and returns whether the user is on a mobile device or tablet. 

With this new set up for mobile detection, we are better able to serve the correct view for users by determining what device they are on and also are able to prevent future bugs that would occur from generating the desktop view at inappropriate times.

## How To Test

The best example of when the app would render a desktop view on mobile is going into the chrome dev tools and selecting the 'iPhone 12 Pro' setting and then switching the orientation to landscape. Before this fix the desktop view would render since the width was bigger than 768px. (!You may need to refresh your screen to correctly see the landscape view the first time. This should not be an issue when actually using an iPhone is related to the dev tools!).

Now the iPhone 12 Pro landscape setting should show a completely mobile view. 

I had to move quite a bit of CSS around to correctly render the mobile view in some instances so it is possible that there maybe something that does not look right. I tried to test everything that I changed as much as possible.
